build_system/seqan3-config.cmake

```diff
@@ -71,7 +71,7 @@
 #
 # ============================================================================
 
-cmake_minimum_required (VERSION 3.2...3.12)
+cmake_minimum_required (VERSION 3.4...3.12)
```

Comment:
is this a maximum version?

Reply:
Basically means: "We are aware of the policy changes introduced in cmake up to version 3.12 and accept the new behaviour of those policies. Please do not throw warnings for new policies."

Otherwise you'd get warnings about policies when compiling with cmake 3.12.

Reply:
Yes, but no in the sense that you might thing. This still works with the most recent cmake version;.

Normally cmake warns you with policy warnings that some default behaviour changed and sometimes you need to set a policy to acknowledge that.

By using this syntax, we say that we tested our script for this version range and I think, that part I'm not to sure, that we acknowledge every policy change in that version range.

This also sets policies to the behaviour of 3.12, s.t. newer cmake versions behave like 3.12.

We introduced that to fix #424.

See https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html?highlight=minimum#command:cmake_minimum_required

Reply:
that's cool :+1: